extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/SegmentGeneratorFrameProcessor.java

```diff
@@ -202,7 +211,14 @@ private void addFrame(final Frame frame)
 
             try {
               rowsWritten++;
+              segmentGenerationProgressCounter.incrementRowProcessedCount();
               appenderator.add(segmentIdWithShardSpec, inputRow, null);
+              if (rowsWritten - lastCounterUpdatedRows > 1000) {
```

Comment:
Lets increment stuff on each added row since we are fetching from an atomic long and the slowest operation would be the merging step anyway.

Reply:
I have made the change. I think this would still end up happening to a small extent, if say some rows are not persisted by the time the last rows are processed.

Reply:
This would be really weird behavior for the end user. 
Should we create a class that extends FireDepartMetric and then update the taskReport as and when relevant calls are made ?

Reply:
Changed!